src/chocolatey/infrastructure.app/commands/ChocolateyCacheCommand.cs

```diff
@@ -0,0 +1,307 @@
+                    }
+                }
+            }
+            else
+            {
+                foreach (var directoryToRemove in _fileSystem.GetDirectories(cacheLocation))
+                {
+                    _fileSystem.DeleteDirectoryChecked(directoryToRemove, recursive: true);
+                }
+            }
+
+            var filesAfterClean = _fileSystem.GetFiles(cacheLocation, "*.dat", SearchOption.AllDirectories);
+
+            if (configuration.CacheCommand.RemoveExpiredItemsOnly)
+            {
+                filesAfterClean = filesAfterClean.Where(f => _fileSystem.GetFileModifiedDate(f) < expirationTimer);
+
+                this.Log().Info("Removed {0} expired cached items in '{1}'", beforeFilesCount - filesAfterClean.Count(), cacheLocation);
+            }
+            else
+            {
+                this.Log().Info("Removed {0} cached items in '{1}'", beforeFilesCount - filesAfterClean.Count(), cacheLocation);
+            }
+        }
+
+        protected override string GetCommandDescription(CommandForAttribute attribute, ChocolateyConfiguration configuration)
+        {
+            return @"Get the statistics of what Chocolatey has cached, or clear any cached
+items in the current context.
+
+This is helpful for when it is necessary for the user to figure out
+when the correct version of a package is not installed, but instead a
+different version is used.";
+        }
+
+        protected override IEnumerable<string> GetCommandExamples(CommandForAttribute[] attributes, ChocolateyConfiguration configuration)
+        {
+            return new[]
+            {
+                "choco cache",
+                "choco cache list",
+                "choco cache remove",
+                "choco cache remove --expired"
+            };
+        }
+
+        protected override IEnumerable<string> GetCommandUsage(CommandForAttribute[] attributes, ChocolateyConfiguration configuration)
+        {
+            return new[] { "choco cache [list]|remove [options/switches]" };
+        }
+        protected virtual void ListCacheStatistic(ChocolateyConfiguration configuration)
+        {
+            var userCacheLocation = ApplicationParameters.HttpCacheUserLocation;
+            var systemCacheLocation = ApplicationParameters.HttpCacheLocation;
+
+            if (userCacheLocation != systemCacheLocation)
+            {
+                this.Log().Info(ChocolateyLoggers.Important, "System Level HTTP Cache");
+                ListCachedItems(configuration, systemCacheLocation);
+
+                this.Log().Info(string.Empty);
+                this.Log().Info(ChocolateyLoggers.Important, "User Level HTTP Cache");
+                ListCachedItems(configuration, userCacheLocation);
+            }
+            else
+            {
+                this.Log().Info(ChocolateyLoggers.Important, "User Level HTTP Cache");
+                ListCachedItems(configuration, userCacheLocation);

```

Comment:
Do we need to use the word Level? Would it not just be as clear, saying 'System HTTP Cache' or 'User HTTP Cache'?

Reply:
I guess we don't really need it, I'll get it the word `level` removed.